sedaprotocol/seda-chain
# Add indexing plugin skeleton

## Motivation

Initial skeleton to guide how future modules can be implemented, as well as SQS connectivity.

## Explanation of Changes

On the boilerplate level the main missing feature is logging. For some reason logs won't show up in the node logs no matter how I configure it.

## Testing

It's a handful but the following steps should help you verify that it works:

0. Build and initialise the chain.
1. Set up SQS emulator. I'd recommend https://github.com/Admiral-Piett/goaws (they provide a docker image) with the following config file:
```yaml
Local:
  Host: sqs
  Region: eu-west-2
  Queues:
    - Name: local-updates.fifo
```
2. Build the plugin with the dev build tag: `go build --tags dev -o build/plugin ./plugins/abci_indexing/plugin.go`.
3. Export the following env variables in a terminal session from which you will start the node.
```sh
export COSMOS_SDK_ABCI=$(readlink -f build/plugin)
export SQS_QUEUE_URL="http://localhost:4100/local-updates.fifo"
export SQS_ENDPOINT=http://localhost:4100
```
4. Edit the `State Streaming` section of the `app.toml` of the node:
```toml
# List of kv store keys to stream out via gRPC.
# The store key names MUST match the module's StoreKey name.
#
# Example:
# ["acc", "bank", "gov", "staking", "mint"[,...]]
# ["*"] to expose all keys.
keys = ["*"]

# The plugin name used for streaming via gRPC.
# Streaming is only enabled if this is set.
# Supported plugins: abci
plugin = "abci"
```
5. Start the node (from the session with the environment variables set).
6. Observe the log output form the SQS emulator, you should start seeing messages entering the queue.

## Related PRs and Issues

Closes: #211
